### ISSUE TITLE
Old legacy NuGet targets are running during builds causing errors/warnings

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#1508]
1. Open \\[internalshare]\public\Jinuj\TestCases\MultiTargetting\MultiTargetting\MultiTargeting.sln
2. Ignore the non-fatal watson and dialog (different [bug ](https://github.com/dotnet/roslyn-project-system/issues/1509)tracking that)

Expected: No errors
Actual: 

```
Error        The package Microsoft.CSharp with version 4.0.1 could not be found in C:\Users\jinuj\.nuget\packages\. Run a NuGet package restore to download the package.    Client    C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Microsoft\NuGet\15.0\Microsoft.NuGet.targets    197    
Warning        Your project is not referencing the ".NETFramework,Version=v4.0" framework. Add a reference to ".NETFramework,Version=v4.0" in the "frameworks" section of your project.json, and then re-run NuGet restore.    Client    C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Microsoft\NuGet\15.0\Microsoft.NuGet.targets    197    
```

**Comment by davkean** [comment]
This is a bit concerning @natidea @srivatsn, what exactly is triggering this this case?

**Comment by natidea** [comment]
Still digging into this, but one observation is that I am only able to repro this when using old style project files with PackageReference-Include-Sdk, and with Import-Common.props and Import-CSharp.targets at the top and bottom of the file respectively instead of just having the Sdk="" attribute.

**Comment by davkean** [comment]
You couldn't repro with the above project?

**Comment by davkean** [comment]
What exactly pulls in these targets? Why would we ever want them?

**Comment by natidea** [comment]
The project above uses the RC2 template with respective Import-Common.props and Import-CSharp.targets. 

Microsoft.NuGet.props/targets are pulled in automatically because they have corresponding files under `%VSINSTALLDIR%\MSBuild\15.0\Imports\Microsoft.Common.Props\ImportBefore` and `%VSINSTALLDIR%\MSBuild\15.0\Microsoft.Common.Targets\ImportAfter` respectively.

**Comment by davkean** [comment]
Normal SDK projects pull in those targets (and Microsoft.NuGet.targets) too:

```
msbuild /pp ConsoleApp1.csproj > targets.log
```


targets.log:
```
[...]
:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Microsoft\NuGet\15.0\Microsoft.NuGet.targets
============================================================================================================================================
-->
  <!--
***********************************************************************************************
Microsoft.NuGet.targets

WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
          created a backup copy.  Incorrect changes to this file will make it
          impossible to load or build your projects from the command-line or the IDE.

Copyright (c) .NET Foundation. All rights reserved. 
***********************************************************************************************
-->
[...]
```

**Comment by natidea** [comment]
Inspecting Sdk.props and Sdk.targets, they import the Common.props/targets, but also import other props/targets that modify properties/items. The generated props/targets for legacy projects that do not have the `Sdk` attribute look different from the build in Sdk.props/targets, at least on cursory inspection - I'll need to take a closer look to see if they are functionally different.

**Comment by davkean** [comment]
@natidea I don't understand. Run the above command that I show above on a normal SDK project from a build today, you can see that Microsoft.NuGet.targets is included in the targets graph. This particular might only be occuring on a project from RC2, but I'm asking why are Microsoft.NuGet.targets included in the graph of SDK projects?

**Comment by srivatsn** [comment]
MSBuild's ImportAfter directory pulls it in from "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Microsoft.Common.Targets\ImportAfter\Microsoft.NuGet.ImportAfter.targets". So all projects will include these targets.

**Comment by natidea** [comment]
I don't think we ever took any steps to exclude Microsoft.NuGet.targets, and it gets included by default when you import Common.props/targets in VS. Instead, we set `<ResolveNuGetPackages>false</ResolveNuGetPackages>` in [Microsoft.NET.Sdk.props](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.props#L87), and that disables a bunch of the targets in Microsoft.NuGet.targets. I don't recall the reasons for this approach, @srivatsn or @dsplaisted may know?

**Comment by natidea** [comment]
Inspecting commit https://github.com/dotnet/sdk/commit/b56a165197d61f2bca30df0036e4da28be2eea4b which added this property, it looks like this is partly a legacy of the way we were selfhosting. Several of our projects needed to explicitly set `<ResolveNuGetPackages>true</ResolveNuGetPackages>` and use the legacy targets.

**Comment by srivatsn** [comment]
Can we add a new property SkipImportingMicrosoftNuGetTargets or some such and skip the import from the file I mentioned above? We can set that property in the SDK.

**Comment by natidea** [comment]
Looks like we can accomplish the same effect by setting several properties to files that don't exist:
```xml
    <NuGetProps>NonExistentFile</NuGetProps>
    <NuGetTargets>NonExistentFile</NuGetTargets>
    <NuGetRestoreTargets>NonExistentFile</NuGetRestoreTargets>
```
When I set this in Microsoft.NET.Sdk.props, Microsoft.NuGet.targets is not loaded (and the assets file has just one RID fixing #1474).

**Comment by natidea** [comment]
It may be too aggressive to include `<NuGetRestoreTargets>NonExistentFile</NuGetRestoreTargets>`. That prevents the inclusion of `%VSINSTALLDIR%\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets`

/cc @emgarten on why we need that file in .NET Core projects. Ultimately, we probably want to match the behavior of the CLI. Searching the dotnet installation, it includes a copy of `Microsoft.NuGet.ImportAfter.targets` with a reference to `NuGetRestoreTargets` but not `NuGetTargets`

**Comment by srivatsn** [comment]
NuGet.targets has the implementation of the restore target to make msbuild /t:restore work.

**Comment by natidea** [comment]
Yep, setting that property breaks msbuild /t:restore :)

**Comment by emgarten** [comment]
> NuGet.targets has the implementation of the restore target to make msbuild /t:restore work.

Yes, you'll need to import it for that reason.

**Comment by natidea** [comment]
Is it also too aggressive to skip Microsoft.NuGet.props. This file only contains a single import: 
```xml
<Import Project="$(MSBuildProjectDirectory)\$(MSBuildProjectName).nuget.props" Condition="..." />
```

Are we including the project's nuget.props file anywhere else?

**Comment by davkean** [comment]
We shoudn't be including that props that's for project.json-based projects.

**Comment by natidea** [comment]
OK, we are importing the project's nuget.props file already in Microsoft.Common.props, so it is not needed:

```xml
<Import Project="$(MSBuildProjectExtensionsPath)$(MSBuildProjectFile).*.props" Condition="..." />
```

**Comment by davkean** [comment]
yes, but be aware that's a different path, that's the intermediate directory, the other you are removing and is not needed is in the project directory.

**Comment by natidea** [comment]
Fixed via https://github.com/dotnet/sdk/issues/870
